### PR TITLE
Support `{{ template-file.yml }}` template variable to reduce LOC

### PR DESCRIPTION
**Pull request by 20k-ultra**
could adapt to support `{{ env(MY-ENV-VAR) }}` too if interested.

```
const envTemplate = (p = "env\(([a-zA-Z-])*\)") => {
  return templateVariable(p);
};
```

Btw, no flowzone file change will be committed since the generated file from the build script is the same.

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -1,21 +1,41 @@
-const fs = require('fs');
-const yaml = require('yaml');
+const fs = require("fs");
+const yaml = require("yaml");
+
+let flowzoneSource = fs.readFileSync("./flowzone.yml", "utf8");
+
+// Pattern to find template variables (sub lines with file)
+const templateVariable = (p) => {
+  return new RegExp(`# {{ *${p} *}}`, "g");
```
Comment (20k-ultra) [inline build.js:8]: Uses a comment so that yaml lsp does not freak out from this template variable

**Comment by 20k-ultra** [comment]
working on the tabbing being respected.

File: build.js [hunk 2]
```diff
@@ -1,21 +1,62 @@
...
+const assert = require("assert");
+const fs = require("fs");
+const yaml = require("yaml");
+
+let flowzoneSource = fs.readFileSync("./flowzone.yml", "utf8");
+
+// Pattern to find template variables (sub lines with file)
+const templateVariable = (p) => {
+  // Tab or spaced comment with curly braces enclosing a variable
+  return new RegExp(`[ \t]*# {{ *${p} *}}`, "g");
```
Comment (20k-ultra) [inline build.js:10]: Uses a comment so YAML lsps do not freak out.

File: build.js [hunk 3]
```diff
@@ -1,21 +1,62 @@
...
+      return tabs(numOfTabs) + l;
+    })
+    .join("\n");
+  function tabs(n, tabReturn = "") {
+    if (n !== 0) {
+      return tabs(n - 1, tabReturn + " ");
+    }
+    return tabReturn + " ";
+  }
+}
```
Comment (20k-ultra) [inline build.js:43]: would have much rather the regex replace to not remove the tabs/spaces. In a regex tester I was able to sub without removing them so it's a JS replace implementation that I cannot figure out how tell not to do replace that way.

File: build.js [hunk 4]
```diff
@@ -1,21 +1,62 @@
...
+    const templateValue = fs.readFileSync(m[1], "utf8");
+    const sub =
+      spacing.length > 0 // Adding template tabs/spaces to template source for proper indenting
+        ? tabText(templateValue, spacing.length)
+        : templateValue;
+    flowzoneSource = flowzoneSource.replaceAll(fileTemplate(m[1]), sub);
+  }
+}
+
+function tabText(text, numOfTabs) {
```
Comment (20k-ultra) [inline build.js:30]: maybe confusing name since yaml does not allow tabs.

Comment (Page-) [reply build.js:30]: yeah, the terminology you want is `indentText` where indentation can be either tabs or space in general (although yaml enforces spaces)

**Comment by 20k-ultra** [comment]
PR should be implemented with https://eemeli.org/yaml/#writing-custom-tags

**Pull request closed without merging**